org_fedora_oscap/gui/spokes/oscap.py

```diff
@@ -388,62 +398,33 @@ def _init_after_data_fetch(self, wait_for):
-        except data_fetch.DataFetchError:
-            self._data_fetch_failed()
-            with self._fetch_flag_lock:
-                self._fetching = False
-            return
-        finally:
-            # stop the spinner in any case
-            fire_gtk_action(self._progress_spinner.stop)
-
-        if self._addon_data.fingerprint:
-            hash_obj = utils.get_hashing_algorithm(self._addon_data.fingerprint)
-            digest = utils.get_file_fingerprint(self._addon_data.raw_preinst_content_path,
-                                                hash_obj)
-            if digest != self._addon_data.fingerprint:
-                self._integrity_check_failed()
-                # fetching done
-                with self._fetch_flag_lock:
-                    self._fetching = False
-                return
+        content_path = None
+        actually_fetched_content = wait_for is not None
 
-        # RPM is an archive at this phase
-        if self._addon_data.content_type in ("archive", "rpm"):
-            # extract the content
-            try:
-                fpaths = common.extract_data(self._addon_data.raw_preinst_content_path,
-                                             common.INSTALLATION_CONTENT_DIR,
-                                             [self._addon_data.content_path])
-            except common.ExtractionError as err:
-                self._extraction_failed(str(err))
-                # fetching done
-                with self._fetch_flag_lock:
-                    self._fetching = False
-                return
+        if actually_fetched_content:
+            content_path = self._addon_data.raw_preinst_content_path
 
-            # and populate missing fields
-            files = content_handling.explore_content_files(fpaths)
-            files = common.strip_content_dir(files)
+        content = self.content_bringer.finish_content_fetch(
+            wait_for, self._addon_data.fingerprint, update_progress_label,
+            content_path, self._handle_error)
+        if not content:
+            with self._fetch_flag_lock:
+                self._fetching = False
 
-            # pylint: disable-msg=E1103
-            self._addon_data.content_path = self._addon_data.content_path or files.xccdf
-            self._addon_data.cpe_path = self._addon_data.cpe_path or files.cpe
-            self._addon_data.tailoring_path = (self._addon_data.tailoring_path or
-                                               files.tailoring)
-        elif self._addon_data.content_type not in (
-                "datastream", "scap-security-guide"):
-            raise common.OSCAPaddonError("Unsupported content type")
+            return
 
         try:
+            if actually_fetched_content:
+                self.content_bringer.use_downloaded_content(content)
+            log.info(f"Opening SCAP content at {self._addon_data.preinst_content_path} with tailoring {self._addon_data.preinst_tailoring_path}")
```

Comment:
I'm sorry but now I realized that self._addon_data.preinst_tailoring_path isn't set when tailoring isn't present so you would get: "Opening SCAP content at /tmp/bzfn with tailoring" in that situation which would be confusing for people reading the log.

Reply:
Fixed by a rebase.